app/services/merge_patient_service.rb

```diff
@@ -74,7 +71,46 @@ def merge_business_identifiers(business_identifier_params, patient)
     end
   end
 
-  def log_update_discarded_patient(merged_patient)
-    NewRelic::Agent.increment_metric("MergePatientService/update_discarded_patient") if merged_patient.discarded?
+  def discarded?(patient)
+    patient.deleted_at.present? && existing_patient&.deleted_at.nil?
+  end
+
+  def discard_patient_data(patient)
+    # Patient has been soft-deleted by the client, server should soft-delete the patient and their associated data.
+    patient.update(deleted_by_user_id: request_metadata[:request_user_id])
```

Comment:
I wouldn't want to touch this code at the moment since it's not a part of the PR, it's just been moved over structurally without logical changes.

However, we typically don't throw exceptions while syncing, errors are silently returned as 200 with an error map and retried by the client eventually (this happens through Validators). 

If there are things that could fail, we should validate them upfront: the `Mergeable` concern takes care of this. The object is already validated at this point. Not convinced that we should be defensive here, esp since this is a single patient out of a batch of patients that would end up failing if we raised wholesale without catching it appropriately.

Reply:
Got it. I think assuming that updates or saves will always work, even after validations, is a dangerous assumption, because there are always other things that can go wrong beyond the individual record's data quality (database is backed up, fk constraints, race conditions, etc).

I agree that tackling that problem piecemeal as part of this PR does not feel warranted, though. 👍